src/Stratis.Bitcoin/Configuration/Settings/ConnectionManagerSettings.cs

```diff
@@ -114,7 +114,7 @@ public ConnectionManagerSettings(NodeSettings nodeSettings)
 
             if (this.ExternalEndpoint == null)
             {
-                this.ExternalEndpoint = new IPEndPoint(IPAddress.Loopback, nodeSettings.Network.DefaultPort);
+                this.ExternalEndpoint = new IPEndPoint(IPAddress.Loopback, port);
```

Comment:
Interesting. How did that work before?

Reply:
It didn't

Reply:
Still not confident that this is the right move.
What was broken before? Is this **definitely** the right value?

Reply:
Maybe we need to do this in another PR.

Reply:
I talked to @fassadlr  who thought the initial code looked like a mistake too.
I don't feel like it shoudl be in another PR, most things in this PR rely on that logic working.

@dangershony , @Aprogiena , @noescape00 , maybe one of you can have a look and let me know if and why I am wrong ? Thanks!

Reply:
This is the thing we have investigated yesterday and the change is good, it is about telling other peers about which port your listening on and we were giving the wrong number so the peers would be confused when they try to connect on wrong port which would fail.

Reply:
I just saw that @bokobza, I thought we talked yesterday, I explained to you I had already shown you a test that was valid and only worked with the change... I also answered your question before (ok very abruptly simply saying it didnt work). I'll resolve the new conflicting file and pray for a merge :)